frontend/public/components/modals/remove-volume-modal.tsx

```diff
@@ -0,0 +1,88 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory';
+import { ContainerSpec, getVolumeType, K8sKind, k8sPatch, K8sResourceKind, Volume, VolumeMount } from '../../module/k8s/';
+import { RowVolumeData } from '../volumes-table';
+
+export const RemoveVolumeModal: React.FC<RemoveVolumeModalProps> = (props) => {
+  const [inProgress, setInProgress] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState('');
+
+  const getRemoveVolumePatch = (resource: K8sResourceKind, rowVolumeData: RowVolumeData) => {
+    const containers: ContainerSpec[] = _.get(resource, 'spec.template.spec.containers', []);
+    const patches = [];
+    let allowRemoveVolume = true;
+    containers.forEach((container: ContainerSpec, i: number) => {
+      const mounts: VolumeMount[] = _.get(container, 'volumeMounts', []);
+      mounts.forEach((mount: VolumeMount, j: number) => {
+        if (mount.name !== rowVolumeData.name) {
+          return;
+        }
+        if (mount.name === rowVolumeData.name) {
```

Comment:
This `if` check is no longer needed with the early return above

Reply:
done